bitsail-connectors/connector-localfilesystem/src/main/java/com/bytedance/bitsail/connector/localfilesystem/reader/LocalFileSystemSourceReader.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2022 Bytedance Ltd. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.bytedance.bitsail.connector.localfilesystem.reader;
+
+import com.bytedance.bitsail.base.connector.reader.v1.SourcePipeline;
+import com.bytedance.bitsail.base.format.DeserializationSchema;
+import com.bytedance.bitsail.common.configuration.BitSailConfiguration;
+import com.bytedance.bitsail.common.row.Row;
+import com.bytedance.bitsail.connector.base.source.SimpleSourceReaderBase;
+import com.bytedance.bitsail.connector.localfilesystem.core.config.FileSystemConfig;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class LocalFileSystemSourceReader extends SimpleSourceReaderBase<Row> {
+  private final FileSystemConfig fileSystemConfig;
+  private final String filePath;
+  private final BufferedReader bufferedReader;
+  private final transient DeserializationSchema<byte[], Row> deserializationSchema;
+
+  public LocalFileSystemSourceReader(BitSailConfiguration jobConf, Context context) {
+    this.fileSystemConfig = new FileSystemConfig(jobConf);
+    this.filePath = fileSystemConfig.getFilePath();
+    this.bufferedReader = loadCsvFile();
+    this.deserializationSchema = DeserializationSchemaFactory.createDeserializationSchema(
+        jobConf,
+        context,
+        fileSystemConfig
+    );
+  }
+
+  private BufferedReader loadCsvFile() {

```

Comment:
this only ready csv file or can read other file too?

Reply:
Nope, it is a function naming negligence. This class can read different files.